src/Microsoft.IdentityModel.Protocols.OpenIdConnect/Configuration/OpenIdConnectConfigurationValidator.cs

```diff
@@ -0,0 +1,107 @@
+    /// </summary>
+    public class OpenIdConnectConfigurationValidator : IConfigurationValidator<OpenIdConnectConfiguration>
+    {
+        private int _minimumNumberOfKeys = DefaultMinimumNumberOfKeys;
+
+        /// <summary>
+        /// 1 is the default minimum number of keys.
+        /// </summary>
+        public static readonly int DefaultMinimumNumberOfKeys = 1;
+
+        /// <summary>
+        /// Validates a OpenIdConnectConfiguration by using current policy.
+        /// </summary>
+        /// <param name="openIdConnectConfiguration">The OpenIdConnectConfiguration to validate.</param>
+        /// <returns>A <see cref="ConfigurationValidationResult"/> that contains validation result.</returns>
+        public ConfigurationValidationResult Validate(OpenIdConnectConfiguration openIdConnectConfiguration)
+        {
+            if (openIdConnectConfiguration == null)
+                throw new ArgumentNullException(nameof(openIdConnectConfiguration));
+
+            if (openIdConnectConfiguration.JsonWebKeySet == null || openIdConnectConfiguration.JsonWebKeySet.Keys.Count == 0)
+            {
+                return new ConfigurationValidationResult
+                {
+                    Exception = new ConfigurationValidationException(LogMessages.IDX21817),
+                    Succeeded = false
+                };
+            }
+
+            if (openIdConnectConfiguration.SigningKeys.Count < MinimumNumberOfKeys)
+            {
+                if (openIdConnectConfiguration.JsonWebKeySet.ConvertKeyInfos.Any())
+                {
+                    return new ConfigurationValidationResult
+                    {
+                        Exception = new ConfigurationValidationException(LogHelper.FormatInvariant(LogMessages.IDX21818, LogHelper.MarkAsNonPII(MinimumNumberOfKeys), string.Join("\n", openIdConnectConfiguration.JsonWebKeySet.ConvertKeyInfos.Select(x => x.Key.Kid.ToString() + ": " + string.Join("; ", x.Value))))),
+                        Succeeded = false
+                    };
+                }
+                else
+                {
+                    return new ConfigurationValidationResult
+                    {
+                        Exception = new ConfigurationValidationException(LogHelper.FormatInvariant(LogMessages.IDX21819, LogHelper.MarkAsNonPII(MinimumNumberOfKeys), LogHelper.MarkAsNonPII(openIdConnectConfiguration.SigningKeys.Count))),
+                        Succeeded = false
+                    };
+                }
+            }
+
+            return new ConfigurationValidationResult
+            {
+                Succeeded = true
+            };
+        }
+
+        /// <summary>
+        /// The minimum number of keys.
+        /// </summary>
+        public int MinimumNumberOfKeys
+        {
+            get { return _minimumNumberOfKeys; }
+            set
+            {
+                if (value < _minimumNumberOfKeys)
```

Comment:
>(value < _minimumNumberOfKeys) [](http://example.com/codeflow?start=19&length=30)

this should just be:

if (value < DefaultMinimumNumberOfKeys)

right? Or do you actually mean that once someone sets it to 2 they can't set it back to 1?

In any case that should match the error message.

Maybe add a unit test for this too then.

Reply:
Updated